Tribler/Test/Community/Tunnel/test_community.py

```diff
@@ -38,24 +38,20 @@ class TestTriblerTunnelCommunity(TestBase):
 
     def setUp(self):
         super(TestTriblerTunnelCommunity, self).setUp()
-        self.private_nodes = []
         self.initialize(TriblerTunnelCommunity, 1)
 
     def tearDown(self):
         super(TestTriblerTunnelCommunity, self).tearDown()
 
-        for node in self.private_nodes:
-            node.unload()
-
     def create_node(self):
         mock_ipv8 = MockIPv8(u"curve25519", TriblerTunnelCommunity, socks_listen_ports=[],
                              exitnode_cache=join(mkdtemp(suffix="_tribler_test_cache"), 'cache.dat'))
         mock_ipv8.overlay.settings.max_circuits = 1
 
         # Load the TrustChain community
-        overlay = TrustChainCommunity(mock_ipv8.my_peer, mock_ipv8.endpoint, mock_ipv8.network,
+        mock_ipv8.trustchain = TrustChainCommunity(mock_ipv8.my_peer, mock_ipv8.endpoint, mock_ipv8.network,
```

Comment:
You can also set `create_trustchain=True` when initializing `MockIPv8`, it does the exact same thing.

Reply:
How was this resolved?

Reply:
(my original answer got swallowed by GitHub smartness...)

If you try to add TrustChain object here by using `create_trustchain=True`, the test errors out:
``` python
Traceback (most recent call last):
  File "/tmp/tribler/Tribler/Test/Community/Tunnel/test_community.py", line 41, in setUp
    self.initialize(TriblerTunnelCommunity, 1)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/test/base.py", line 37, in initialize
    self.nodes = [self.create_node(*args, **kwargs) for _ in range(node_count)]
  File "/tmp/tribler/Tribler/Test/Community/Tunnel/test_community.py", line 49, in create_node
    exitnode_cache=join(mkdtemp(suffix="_tribler_test_cache"), 'cache.dat'))
  File "/tmp/tribler/Tribler/pyipv8/ipv8/test/mocking/ipv8.py", line 40, in __init__
    self.overlay = overlay_class(self.my_peer, self.endpoint, self.network, *args, **kwargs)
  File "/tmp/tribler/Tribler/community/triblertunnel/community.py", line 45, in __init__
    super(TriblerTunnelCommunity, self).__init__(*args, **kwargs)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/messaging/anonymization/hidden_services.py", line 25, in __init__
    super(HiddenTunnelCommunity, self).__init__(*args, **kwargs)
  File "/tmp/tribler/Tribler/pyipv8/ipv8/messaging/anonymization/community.py", line 154, in __init__
    super(TunnelCommunity, self).__init__(*args, **kwargs)
exceptions.TypeError: __init__() got an unexpected keyword argument 'trustchain'

```

Reply:
@ichorid you need `create_trustchain` instead of `trustchain`:

https://github.com/Tribler/py-ipv8/blob/1caefbf2c7654c9fb2301b5ad9243877f377ae2a/ipv8/test/mocking/ipv8.py#L15

Reply:
Aha, I see. Because of this line: https://github.com/Tribler/py-ipv8/blob/1caefbf2c7654c9fb2301b5ad9243877f377ae2a/ipv8/test/mocking/ipv8.py#L32